src/intern.rs

```diff
@@ -6,6 +6,10 @@ extern {
     pub fn rb_ary_new_capa(capacity: c_long) -> VALUE;
     pub fn rb_ary_push(array: VALUE, item: VALUE) -> VALUE;
 
+    pub fn rb_hash_new() -> VALUE;
+    pub fn rb_hash_aset(hash: VALUE, key: VALUE, val: VALUE) -> VALUE;
+    pub fn rb_hash_foreach(hash: VALUE, func: extern "C" fn(key: VALUE, val: VALUE, farg: VALUE), farg: VALUE);
```

Comment:
This is not quite right. At minimum `func` should be typed as returning a `c_int`, although the header actually typed it as an `ANYARGS<c_int>`. I agree that we _probably_ want to be more precise here and match the actual `rb_foreach_func` definition (should mention that in the notes section of the documentation).

Reply:
UB: when the func returns anything other than the `ST_*` values

Reply:
Yeah, I copied some of this from Helix which I think does it wrong. For reference:
```c
typedef int rb_foreach_func(VALUE, VALUE, VALUE);
```

As for `ST_*` should we just make our own version of the enum?